TradeItIosTicketSDK2/TradeItLinkedBrokerCache.swift

```diff
@@ -139,37 +186,89 @@ class TradeItLinkedBrokerCache {
+            serializedAccount[ACCOUNT_NAME_KEY] = encryptData(account.accountName)
+            serializedAccount[ACCOUNT_NUMBER_KEY] = encryptData(account.accountNumber)
+            serializedAccount[ACCOUNT_BASE_CURRENCY_KEY] = encryptData(account.accountBaseCurrency)
+            serializedAccount[ACCOUNT_ENABLED_KEY] = encryptData(account.isEnabled ? ACCOUNT_ENABLED_VALUE : ACCOUNT_DISABLED_VALUE)
 
             if let balance = account.balance,
                 let buyingPower = balance.buyingPower {
 
                 if let balanceLastUpdated = account.balanceLastUpdated {
-                    serializedAccount[BALANCE_LAST_UPDATED_KEY] = balanceLastUpdated
+                    serializedAccount[BALANCE_LAST_UPDATED_KEY] = encryptData(TradeItLinkedBrokerCache.dateFormatter.string(from: balanceLastUpdated))
                 }
                 
-                serializedAccount[BALANCE_BUYING_POWER_KEY] = buyingPower
+                serializedAccount[BALANCE_BUYING_POWER_KEY] = encryptData(TradeItLinkedBrokerCache.numberFormatter.string(from: buyingPower))
                 
                 let buyingPowerLabel = balance.buyingPowerLabel ?? "Buying Power"
-                serializedAccount[BALANCE_BUYING_POWER_LABEL_KEY] = buyingPowerLabel
+                serializedAccount[BALANCE_BUYING_POWER_LABEL_KEY] = encryptData(buyingPowerLabel)
             }
 
             if let fxBalance = account.fxBalance,
                 let buyingPowerBaseCurrency = fxBalance.buyingPowerBaseCurrency {
 
                 if let balanceLastUpdated = account.balanceLastUpdated {
-                    serializedAccount[BALANCE_LAST_UPDATED_KEY] = balanceLastUpdated
+                    serializedAccount[BALANCE_LAST_UPDATED_KEY] = encryptData(TradeItLinkedBrokerCache.dateFormatter.string(from: balanceLastUpdated))
                 }
 
-                serializedAccount[FX_BALANCE_BUYING_POWER_KEY] = buyingPowerBaseCurrency
+                serializedAccount[FX_BALANCE_BUYING_POWER_KEY] = encryptData(TradeItLinkedBrokerCache.numberFormatter.string(from: buyingPowerBaseCurrency))
             }
 
             serializeAccountsList.append(serializedAccount)
         }
         
         return serializeAccountsList
     }
+    
+    private func encryptData(_ data: String?) -> String {
+        var encryptedData = data ?? ""
+        do {
+            if let data = data {
+                encryptedData = try self.aesKey.encrypt(Array(data.utf8)).toBase64()!
+            }
+        } catch {
+            print("===> ERROR encryptData: \(error)")
+        }
+        return encryptedData
+    }
+    
+    private func decryptData(_ data: String?) -> String {
+        var decryptedData = data ?? ""
+        do {
+            if let data = data {
+                decryptedData = try data.decryptBase64ToString(cipher: self.aesKey)
+            }
+        } catch {
+            print("===> ERROR decryptData: \(error)")
```

Comment:
For existing app, it will just show the error in the console and by default return the non encrypted data, so no regression. The next time there is a refresh, the data will be encrypted.